astropy/table/tests/test_operations.py

```diff
@@ -961,6 +961,105 @@ def test_mixin_functionality(self, mixin_cols):
                     'vstack unavailable' in str(err.value))
 
 
+class TestCStack():
+
+    def _setup(self, t_cls=Table):
+        self.t1 = t_cls.read([' a   b',
+                              ' 0. foo',
+                              ' 1. bar'], format='ascii')
+
+        self.t2 = t_cls.read([' a    b   c',
+                              ' 2.  pez  4',
+                              ' 3.  sez  5'], format='ascii')
+
+        self.t3 = t_cls({'a': [[5., 6.], [4., 3.]],
+                         'b': [['foo', 'bar'], ['pez', 'sez']]},
+                        names=('a', 'b'))
+        self.t4 = t_cls(self.t1, copy=True, masked=t_cls is Table)
+        self.t5 = t_cls({'a': [[4., 2.], [1., 6.]],
+                         'b': [['foo', 'pez'], ['bar', 'sez']]},
+                        names=('a', 'b'))
+        self.t6 = t_cls.read([' a    b   c',
+                              ' 7.  pez  2',
+                              ' 4.  sez  6',
+                              ' 6.  foo  3'], format='ascii')
+
+    def test_stack_table_column(self, operation_table_type):
+        self._setup(operation_table_type)
+        t2 = self.t1.copy()
+        out = table.cstack([self.t1, t2['a']])
+        assert out.pformat() == ['  a [2]      b [2]  ',
+                                 '---------- ---------',
+                                 '0.0 .. 0.0 foo .. --',
+                                 '1.0 .. 1.0 bar .. --']
+
+    def test_stack_basic_outer(self, operation_table_type):
+            if operation_table_type is QTable:
+                pytest.xfail('Quantity columns do not support masking.')
+            self._setup(operation_table_type)
+            t1 = self.t1
+            t2 = self.t2
+            t4 = self.t4
+            t12 = table.cstack([t1, t2], join_type='outer')
+            assert t12.pformat() == ['  a [2]      b [2]     c [2] ',
+                                     '---------- ---------- -------',
+                                     '0.0 .. 2.0 foo .. pez -- .. 4',
+                                     '1.0 .. 3.0 bar .. sez -- .. 5']
+            t124 = table.cstack([t1, t2, t4], join_type='outer')
+            assert t124.pformat() == ['  a [3]      b [3]     c [3]  ',
+                                      '---------- ---------- --------',
+                                      '0.0 .. 0.0 foo .. foo -- .. --',
+                                      '1.0 .. 1.0 bar .. bar -- .. --']
+
+    def test_stack_basic_inner(self, operation_table_type):
+        self._setup(operation_table_type)
+        t1 = self.t1
+        t2 = self.t2
+        t4 = self.t4
+
+        t12 = table.cstack([t1, t2], join_type='inner')
```

Comment:
Same comment as before, delete the `t12` test.

Reply:
Done